src/widget/wtrackmenu.cpp

```diff
@@ -138,6 +139,166 @@ void WTrackMenu::createMenus() {
         m_pClearMetadataMenu->setTitle(tr("Reset"));
     }
+
+    if (featureIsEnabled(Feature::SearchRelated)) {
+        // TODO: Create a new subclass of QMenu?
+        DEBUG_ASSERT(!m_pSearchRelatedMenu);
+        m_pSearchRelatedMenu =
+                make_parented<QMenu>(tr("Search related Tracks"), this);
+        connect(m_pSearchRelatedMenu,
+                &QMenu::aboutToShow,
+                this,
+                [this] {
+                    m_pSearchRelatedMenu->clear();
+                    m_pSearchRelatedMenu->setEnabled(false);
+                    const auto pTrack = getFirstTrackPointer();
+                    if (!pTrack) {
+                        return;
+                    }
+                    const auto musicalKey = pTrack->getKeyText();
+                    if (!musicalKey.isEmpty()) {
+                        const auto musicalKeyText =
+                                mixxx::escapeTextPropertyWithoutShortcuts(musicalKey);
+                        m_pSearchRelatedMenu->addAction(
+                                tr("Harmonic keys for \"%1\"")
+                                        .arg(musicalKeyText),
+                                [this, musicalKey]() {
+                                    m_pLibrary->searchTracksInCollection(
+                                            QStringLiteral("~key:\"") +
+                                            musicalKey + QChar('"'));
+                                });
+                    }
+                    const auto genre = pTrack->getGenre();
+                    if (!genre.isEmpty()) {
+                        const auto genreText =
+                                mixxx::escapeTextPropertyWithoutShortcuts(genre);
+                        m_pSearchRelatedMenu->addAction(
+                                tr("Genre \"%1\"")
+                                        .arg(genreText),
+                                [this, genre]() {
+                                    m_pLibrary->searchTracksInCollection(
+                                            QStringLiteral("genre:\"") +
+                                            genre + QChar('"'));
+                                });
+                    }
+                    const auto title = pTrack->getTitle();
+                    if (!title.isEmpty()) {
+                        const auto titleText =
+                                mixxx::escapeTextPropertyWithoutShortcuts(title);
+                        m_pSearchRelatedMenu->addAction(
+                                tr("Title \"%1\"")
+                                        .arg(titleText),
+                                [this, title]() {
+                                    m_pLibrary->searchTracksInCollection(
+                                            QStringLiteral("title:\"") +
+                                            title + QChar('"'));
+                                });
+                    }
```

Comment:
At first I questioned how this could be useful. Then I realized it could be used to search for remixes of the track. However, this use case isn't very obvious. Any ideas for making it more discoverable? Also, it doesn't work as well for finding similar tracks from a remix unless I delete the `(REMIXER Remix)` suffix manually.

Reply:
Same here. I noticed how convenient this is when searching for duplicates or cover versions of a track. Manually editing the prepared search query is also an option. Easier than copy&paste a title query manually.